[Bug] Class conflict when stopping application and doing savepoint

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### Java Version

1.8

### Scala Version

2.12.x

### StreamPark Version

2.1.0

### Flink Version

1.16.0

### deploy mode

None

### What happened

Class conflict when stopping application and doing savepoint

### Error Exception

```log
java.util.concurrent.CompletionException: java.lang.reflect.InvocationTargetException
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1606)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.streampark.flink.client.FlinkClient$.$anonfun$proxy$1(FlinkClient.scala:68)
	at org.apache.streampark.flink.proxy.FlinkShimsProxy$.$anonfun$proxy$1(FlinkShimsProxy.scala:79)
	at org.apache.streampark.common.util.ClassLoaderUtils$.runAsClassLoader(ClassLoaderUtils.scala:40)
	at org.apache.streampark.flink.proxy.FlinkShimsProxy$.proxy(FlinkShimsProxy.scala:79)
	at org.apache.streampark.flink.client.FlinkClient$.proxy(FlinkClient.scala:63)
	at org.apache.streampark.flink.client.FlinkClient$.cancel(FlinkClient.scala:47)
	at org.apache.streampark.flink.client.FlinkClient.cancel(FlinkClient.scala)
	at org.apache.streampark.console.core.service.impl.ApplicationServiceImpl.lambda$cancel$6(ApplicationServiceImpl.java:1251)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1604)
	... 3 more
Caused by: java.lang.NoSuchMethodError: org.apache.flink.client.program.ClusterClient.cancelWithSavepoint(Lorg/apache/flink/api/common/JobID;Ljava/lang/String;)Ljava/util/concurrent/CompletableFuture;
	at org.apache.streampark.flink.core.FlinkClientTrait.cancelWithSavepoint(FlinkClientTrait.scala:32)
	at org.apache.streampark.flink.client.trait.FlinkClientTrait.cancelJob(FlinkClientTrait.scala:486)
	at org.apache.streampark.flink.client.trait.FlinkClientTrait.cancelJob$(FlinkClientTrait.scala:474)
	at org.apache.streampark.flink.client.impl.YarnPerJobClient$.org$apache$streampark$flink$client$trait$YarnClientTrait$$super$cancelJob(YarnPerJobClient.scala:41)
	at org.apache.streampark.flink.client.trait.YarnClientTrait.$anonfun$doCancel$1(YarnClientTrait.scala:72)
	at org.apache.streampark.flink.client.trait.YarnClientTrait.$anonfun$executeClientAction$1(YarnClientTrait.scala:56)
	at scala.util.Try$.apply(Try.scala:209)
	at org.apache.streampark.flink.client.trait.YarnClientTrait.executeClientAction(YarnClientTrait.scala:56)
	at org.apache.streampark.flink.client.trait.YarnClientTrait.doCancel(YarnClientTrait.scala:71)
	at org.apache.streampark.flink.client.trait.YarnClientTrait.doCancel$(YarnClientTrait.scala:70)
	at org.apache.streampark.flink.client.impl.YarnPerJobClient$.doCancel(YarnPerJobClient.scala:123)
	at org.apache.streampark.flink.client.trait.FlinkClientTrait.cancel(FlinkClientTrait.scala:179)
	at org.apache.streampark.flink.client.trait.FlinkClientTrait.cancel$(FlinkClientTrait.scala:163)
	at org.apache.streampark.flink.client.impl.YarnPerJobClient$.cancel(YarnPerJobClient.scala:41)
	at org.apache.streampark.flink.client.FlinkClientHandler$.cancel(FlinkClientHandler.scala:47)
	at org.apache.streampark.flink.client.FlinkClientHandler.cancel(FlinkClientHandler.scala)
	... 16 more
```


### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!(您是否要贡献这个PR?)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/streampark-common/src/main/scala/org/apache/streampark/common/conf/FlinkVersion.scala
+++ b/streampark-common/src/main/scala/org/apache/streampark/common/conf/FlinkVersion.scala
@@ -109,6 +109,18 @@ class FlinkVersion(val flinkHome: String) extends java.io.Serializable with Logg
     distJar.head
   }
 
+  def checkVersion(throwable: Boolean): Boolean = {
+    version.split("\\.").map(_.trim.toInt) match {
+      case Array(1, v, _) if v >= 12 && v <= 17 => true
+      case _ =>
+        if (throwable) {
+          throw new UnsupportedOperationException(s"Unsupported flink version: $version")
+        } else {
+          false
+        }
+    }
+  }
+
   // StreamPark flink shims version, like "streampark-flink-shims_flink-1.13"
   lazy val shimsVersion: String = s"streampark-flink-shims_flink-$majorVersion"
 

--- a/streampark-common/src/main/scala/org/apache/streampark/common/conf/FlinkVersion.scala
+++ b/streampark-common/src/main/scala/org/apache/streampark/common/conf/FlinkVersion.scala
@@ -109,11 +109,11 @@ class FlinkVersion(val flinkHome: String) extends java.io.Serializable with Logg
     distJar.head
   }
 
-  def checkVersion(throwable: Boolean = true): Boolean = {
+  def checkVersion(throwException: Boolean = true): Boolean = {
     version.split("\\.").map(_.trim.toInt) match {
       case Array(1, v, _) if v >= 12 && v <= 17 => true
       case _ =>
-        if (throwable) {
+        if (throwException) {
           throw new UnsupportedOperationException(s"Unsupported flink version: $version")
         } else {
           false

--- a/streampark-common/src/main/scala/org/apache/streampark/common/conf/FlinkVersion.scala
+++ b/streampark-common/src/main/scala/org/apache/streampark/common/conf/FlinkVersion.scala
@@ -109,7 +109,7 @@ class FlinkVersion(val flinkHome: String) extends java.io.Serializable with Logg
     distJar.head
   }
 
-  def checkVersion(throwable: Boolean): Boolean = {
+  def checkVersion(throwable: Boolean = true): Boolean = {
     version.split("\\.").map(_.trim.toInt) match {
       case Array(1, v, _) if v >= 12 && v <= 17 => true
       case _ =>

--- a/streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/controller/ApplicationBuildPipelineController.java
+++ b/streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/controller/ApplicationBuildPipelineController.java
@@ -25,10 +25,12 @@ import org.apache.streampark.console.core.bean.AppBuildDockerResolvedDetail;
 import org.apache.streampark.console.core.entity.AppBuildPipeline;
 import org.apache.streampark.console.core.entity.Application;
 import org.apache.streampark.console.core.entity.ApplicationLog;
+import org.apache.streampark.console.core.entity.FlinkEnv;
 import org.apache.streampark.console.core.enums.Operation;
 import org.apache.streampark.console.core.service.AppBuildPipeService;
 import org.apache.streampark.console.core.service.ApplicationLogService;
 import org.apache.streampark.console.core.service.ApplicationService;
+import org.apache.streampark.console.core.service.FlinkEnvService;
 import org.apache.streampark.console.core.service.FlinkSqlService;
 import org.apache.streampark.flink.packer.pipeline.DockerResolvedSnapshot;
 import org.apache.streampark.flink.packer.pipeline.PipelineType;
@@ -66,6 +68,8 @@ public class ApplicationBuildPipelineController {
 
   @Autowired private ApplicationLogService applicationLogService;
 
+  @Autowired private FlinkEnvService flinkEnvService;
+
   /**
    * Release application building pipeline.
    *
@@ -100,11 +104,15 @@ public class ApplicationBuildPipelineController {
     try {
       Application app = applicationService.getById(appId);
 
-      ApplicationLog applicationLog = new ApplicationLog();
-      applicationLog.setOptionName(Operation.RELEASE.getValue());
-      applicationLog.setAppId(app.getId());
-      applicationLog.setOptionTime(new Date());
+      // 1) check flink version
+      FlinkEnv env = flinkEnvService.getById(app.getVersionId());
+      boolean checkVersion = env.getFlinkVersion().checkVersion(false);
+      if (!checkVersion) {
+        throw new ApiAlertException(
+            "Unsupported flink version: " + env.getFlinkVersion().version());
+      }
 
+      // 2) check env
       boolean envOk = applicationService.checkEnv(app);
       if (!envOk) {
         throw new ApiAlertException(
@@ -118,6 +126,12 @@ public class ApplicationBuildPipelineController {
       // check if you need to go through the build process (if the jar and pom have changed,
       // you need to go through the build process, if other common parameters are modified,
       // you don't need to go through the build process)
+
+      ApplicationLog applicationLog = new ApplicationLog();
+      applicationLog.setOptionName(Operation.RELEASE.getValue());
+      applicationLog.setAppId(app.getId());
+      applicationLog.setOptionTime(new Date());
+
       boolean needBuild = applicationService.checkBuildAndUpdate(app);
       if (!needBuild) {
         applicationLog.setSuccess(true);

--- a/streampark-flink/streampark-flink-client/streampark-flink-client-api/src/main/scala/org/apache/streampark/flink/client/FlinkClient.scala
+++ b/streampark-flink/streampark-flink-client/streampark-flink-client-api/src/main/scala/org/apache/streampark/flink/client/FlinkClient.scala
@@ -60,6 +60,7 @@ object FlinkClient extends Logger {
   }
 
   private[this] def proxy[T: ClassTag](request: Object, flinkVersion: FlinkVersion, requestBody: (String, String)): T = {
+    flinkVersion.checkVersion()
     FlinkShimsProxy.proxy(flinkVersion, (classLoader: ClassLoader) => {
       val submitClass = classLoader.loadClass(FLINK_CLIENT_HANDLER_CLASS_NAME)
       val requestClass = classLoader.loadClass(requestBody._1)

--- a/streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/YarnApplicationClient.scala
+++ b/streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/YarnApplicationClient.scala
@@ -67,11 +67,7 @@ object YarnApplicationClient extends YarnClientTrait {
         submitRequest.hdfsWorkspace.appPlugins)
       submitRequest.developmentMode match {
         case DevelopmentMode.FLINK_SQL =>
-          val version = submitRequest.flinkVersion.version.split("\\.").map(_.trim.toInt)
-          version match {
-            case Array(1, v, _) if v >= 12 && v <= 17 => array += s"${workspace.APP_SHIMS}/flink-1.$v"
-            case _ => throw new UnsupportedOperationException(s"Unsupported flink version: ${submitRequest.flinkVersion}")
-          }
+          array += s"${workspace.APP_SHIMS}/flink-${submitRequest.flinkVersion.majorVersion}"
           val jobLib = s"${workspace.APP_WORKSPACE}/${submitRequest.id}/lib"
           if (HdfsUtils.exists(jobLib)) {
             array += jobLib

--- a/streampark-flink/streampark-flink-packer/src/main/scala/org/apache/streampark/flink/packer/pipeline/BuildRequest.scala
+++ b/streampark-flink/streampark-flink-packer/src/main/scala/org/apache/streampark/flink/packer/pipeline/BuildRequest.scala
@@ -55,13 +55,7 @@ sealed trait FlinkBuildParam extends BuildParam {
       localWorkspace.APP_PLUGINS,
       customFlinkUserJar)
     if (developmentMode == DevelopmentMode.FLINK_SQL) {
-      providedLibs += {
-        val version = flinkVersion.version.split("\\.").map(_.trim.toInt)
-        version match {
-          case Array(1, v, _) if v >= 12 && v <= 17 => s"${localWorkspace.APP_SHIMS}/flink-1.$v"
-          case _ => throw new UnsupportedOperationException(s"Unsupported flink version: $flinkVersion")
-        }
-      }
+      providedLibs += s"${localWorkspace.APP_SHIMS}/flink-${flinkVersion.majorVersion}"
     }
     dependencyInfo.merge(providedLibs.toSet)
   }

--- a/streampark-flink/streampark-flink-proxy/src/main/scala/org/apache/streampark/flink/proxy/ChildFirstClassLoader.scala
+++ b/streampark-flink/streampark-flink-proxy/src/main/scala/org/apache/streampark/flink/proxy/ChildFirstClassLoader.scala
@@ -34,7 +34,7 @@ import scala.util.Try
  */
 
 class ChildFirstClassLoader(urls: Array[URL], parent: ClassLoader, flinkResourcePattern: Pattern, classLoadingExceptionHandler: Consumer[Throwable])
-    extends URLClassLoader(urls, parent) {
+  extends URLClassLoader(urls, parent) {
 
   ClassLoader.registerAsParallelCapable()
 
@@ -66,10 +66,9 @@ class ChildFirstClassLoader(urls: Array[URL], parent: ClassLoader, flinkResource
 
   @throws[ClassNotFoundException]
   override def loadClass(name: String, resolve: Boolean): Class[_] = {
-    try this.synchronized {
-        return this.loadClassWithoutExceptionHandling(name, resolve)
-      }
-    catch {
+    try {
+      this.synchronized(this.loadClassWithoutExceptionHandling(name, resolve))
+    } catch {
       case e: Throwable =>
         classLoadingExceptionHandler.accept(e)
         throw e

--- a/streampark-flink/streampark-flink-proxy/src/main/scala/org/apache/streampark/flink/proxy/FlinkShimsProxy.scala
+++ b/streampark-flink/streampark-flink-proxy/src/main/scala/org/apache/streampark/flink/proxy/FlinkShimsProxy.scala
@@ -86,7 +86,7 @@ object FlinkShimsProxy extends Logger {
 
         // 3) add only streampark shims jar
         addShimsUrls(flinkVersion, file => {
-          if (file.getName.startsWith(FLINK_SHIMS_PREFIX)) {
+          if (file.getName.startsWith("streampark-flink-shims")) {
             shimsUrls += file.toURI.toURL
           }
         })

--- a/streampark-flink/streampark-flink-proxy/src/main/scala/org/apache/streampark/flink/proxy/FlinkShimsProxy.scala
+++ b/streampark-flink/streampark-flink-proxy/src/main/scala/org/apache/streampark/flink/proxy/FlinkShimsProxy.scala
@@ -29,15 +29,13 @@ import org.apache.streampark.common.util.{ClassLoaderUtils, Logger, Utils}
 
 object FlinkShimsProxy extends Logger {
 
+  private[this] val SHIMS_CLASS_LOADER_CACHE = MutableMap[String, ClassLoader]()
+
+  private[this] val VERIFY_SQL_CLASS_LOADER_CACHE = MutableMap[String, ClassLoader]()
+
   private[this] val INCLUDE_PATTERN: Pattern = Pattern.compile(
     "(json4s|jackson)(.*).jar",
     Pattern.CASE_INSENSITIVE | Pattern.DOTALL)
-
-  private[this] val SHIMS_CLASS_LOADER_CACHE = MutableMap[String, ClassLoader]()
-
-  private[this] val SHIMS_PATTERN_CACHE = MutableMap[String, Pattern]()
-
-  private[this] val VERIFY_SQL_CLASS_LOADER_CACHE = MutableMap[String, ClassLoader]()
 
   private[this] def getFlinkShimsResourcePattern(flinkLargeVersion: String) =
     Pattern.compile(
@@ -47,24 +45,6 @@ object FlinkShimsProxy extends Logger {
   private[this] def getStreamParkLibPattern(scalaVersion: String): Pattern = Pattern.compile(
     s"streampark-(.*)_$scalaVersion-(.*).jar",
     Pattern.CASE_INSENSITIVE | Pattern.DOTALL)
-
-  private[this] def getShimsLibPattern(flinkVersion: FlinkVersion): Pattern = {
-    flinkVersion.version.split("\\.").map(_.trim.toInt) match {
-      case Array(1, v, _) =>
-        SHIMS_PATTERN_CACHE.getOrElseUpdate(s"1.$v", {
-          val regexp = if (v >= 12 && v <= 14) {
-            s"streampark-flink-shims_flink-1.${v}_(2.11|2.12)-(.*).jar"
-          } else if (v >= 15) {
-            s"streampark-flink-shims_flink-1.${v}_2.12-(.*).jar"
-          } else {
-            throw new UnsupportedOperationException(s"Unsupported flink version: $flinkVersion")
-          }
-          Pattern.compile(regexp, Pattern.CASE_INSENSITIVE | Pattern.DOTALL)
-        })
-      case _ =>
-        throw new UnsupportedOperationException(s"Unsupported flink version: $flinkVersion")
-    }
-  }
 
   /**
    * Get shimsClassLoader to execute for scala API
@@ -92,7 +72,6 @@ object FlinkShimsProxy extends Logger {
     ClassLoaderUtils.runAsClassLoader[T](shimsClassLoader, () => func(shimsClassLoader))
   }
 
-  // flink 1.12 1.13~1.14 1.15 1.16 parseSql class exist in different dependencies,
   // need to load all flink-table dependencies compatible with different versions
   def getVerifySqlLibClassLoader(flinkVersion: FlinkVersion): ClassLoader = {
     logInfo(s"add verify sql lib,flink version: $flinkVersion")
@@ -107,13 +86,12 @@ object FlinkShimsProxy extends Logger {
         val shimsUrls = ListBuffer[URL](libTableURL ++ optTableURL: _*)
 
         // 3) add only streampark shims jar
-        addShimsUrls(
-          flinkVersion,
-          file => {
-            if (file != null && file.getName.startsWith("streampark-flink-shims")) {
-              shimsUrls += file.toURI.toURL
-            }
-          })
+        addShimsUrls(flinkVersion, file => {
+          if (file != null && file.getName.startsWith("streampark-flink-shims")) {
+            shimsUrls += file.toURI.toURL
+          }
+        })
+
         new ChildFirstClassLoader(
           shimsUrls.toArray,
           Thread.currentThread().getContextClassLoader,
@@ -127,29 +105,25 @@ object FlinkShimsProxy extends Logger {
 
     val libPath = new File(s"$appHome/lib")
     require(libPath.exists())
+
     val majorVersion = flinkVersion.majorVersion
     val scalaVersion = flinkVersion.scalaVersion
     val streamParkMatcher = getStreamParkLibPattern(scalaVersion)
 
     libPath.listFiles().foreach((jar: File) => {
-      try {
-        val shimsMatcher = getShimsLibPattern(flinkVersion).matcher(jar.getName)
-        if (shimsMatcher.matches()) {
-          if (majorVersion == shimsMatcher.group(1) && scalaVersion == shimsMatcher.group(2)) {
-            addShimUrl(jar)
-          }
-        } else {
-          if (INCLUDE_PATTERN.matcher(jar.getName).matches()) {
-            addShimUrl(jar)
-            logInfo(s"include jar lib: ${jar.getName}")
-          }
-          if (streamParkMatcher.matcher(jar.getName).matches()) {
-            addShimUrl(jar)
-            logInfo(s"include streampark lib: ${jar.getName}")
-          }
+      val shimsPrefix = s"streampark-flink-shims_flink-${majorVersion}_$scalaVersion"
+      if (jar.getName.startsWith(shimsPrefix) && jar.getName.endsWith(".jar")) {
+        addShimUrl(jar)
+        logInfo(s"include flink shims jar lib: ${jar.getName}")
+      } else {
+        if (INCLUDE_PATTERN.matcher(jar.getName).matches()) {
+          addShimUrl(jar)
+          logInfo(s"include jar lib: ${jar.getName}")
         }
-      } catch {
-        case e: Exception => e.printStackTrace()
+        if (streamParkMatcher.matcher(jar.getName).matches()) {
+          addShimUrl(jar)
+          logInfo(s"include streampark lib: ${jar.getName}")
+        }
       }
     })
   }
@@ -177,13 +151,11 @@ object FlinkShimsProxy extends Logger {
         val shimsUrls = ListBuffer[URL](libURL: _*)
 
         // 2) add all shims jar
-        addShimsUrls(
-          flinkVersion,
-          file => {
-            if (file != null) {
-              shimsUrls += file.toURI.toURL
-            }
-          })
+        addShimsUrls(flinkVersion, file => {
+          if (file != null) {
+            shimsUrls += file.toURI.toURL
+          }
+        })
 
         new ChildFirstClassLoader(
           shimsUrls.toArray,
@@ -198,8 +170,7 @@ object FlinkShimsProxy extends Logger {
     file.listFiles.filter(filterFun).map(_.toURI.toURL).toList
   }
 
-  @throws[Exception]
-  def getObject[T](loader: ClassLoader, obj: Object): T = {
+  @throws[Exception] def getObject[T](loader: ClassLoader, obj: Object): T = {
     val arrayOutputStream = new ByteArrayOutputStream
     val result = Utils.tryWithResource(new ObjectOutputStream(arrayOutputStream))(objectOutputStream => {
       objectOutputStream.writeObject(obj)

--- a/streampark-flink/streampark-flink-proxy/src/main/scala/org/apache/streampark/flink/proxy/FlinkShimsProxy.scala
+++ b/streampark-flink/streampark-flink-proxy/src/main/scala/org/apache/streampark/flink/proxy/FlinkShimsProxy.scala
@@ -42,9 +42,8 @@ object FlinkShimsProxy extends Logger {
       s"flink-(.*)-$flinkLargeVersion(.*).jar",
       Pattern.CASE_INSENSITIVE | Pattern.DOTALL)
 
-  private[this] def getStreamParkLibPattern(scalaVersion: String): Pattern = Pattern.compile(
-    s"streampark-(.*)_$scalaVersion-(.*).jar",
-    Pattern.CASE_INSENSITIVE | Pattern.DOTALL)
+
+  private[this] lazy val FLINK_SHIMS_PREFIX = "streampark-flink-shims_flink"
 
   /**
    * Get shimsClassLoader to execute for scala API
@@ -108,21 +107,24 @@ object FlinkShimsProxy extends Logger {
 
     val majorVersion = flinkVersion.majorVersion
     val scalaVersion = flinkVersion.scalaVersion
-    val streamParkMatcher = getStreamParkLibPattern(scalaVersion)
 
     libPath.listFiles().foreach((jar: File) => {
-      val shimsPrefix = s"streampark-flink-shims_flink-${majorVersion}_$scalaVersion"
-      if (jar.getName.startsWith(shimsPrefix) && jar.getName.endsWith(".jar")) {
-        addShimUrl(jar)
-        logInfo(s"include flink shims jar lib: ${jar.getName}")
-      } else {
-        if (INCLUDE_PATTERN.matcher(jar.getName).matches()) {
-          addShimUrl(jar)
-          logInfo(s"include jar lib: ${jar.getName}")
-        }
-        if (streamParkMatcher.matcher(jar.getName).matches()) {
-          addShimUrl(jar)
-          logInfo(s"include streampark lib: ${jar.getName}")
+      val jarName = jar.getName
+      if (jarName.endsWith(".jar")) {
+        if (jarName.startsWith(FLINK_SHIMS_PREFIX)) {
+          val prefixVer = s"$FLINK_SHIMS_PREFIX-${majorVersion}_$scalaVersion"
+          if (jarName.startsWith(prefixVer)) {
+            addShimUrl(jar)
+            logInfo(s"include flink shims jar lib: $jarName")
+          }
+        } else {
+          if (INCLUDE_PATTERN.matcher(jarName).matches()) {
+            addShimUrl(jar)
+            logInfo(s"include jar lib: $jarName")
+          } else if (jarName.matches(s"^streampark-.*_$scalaVersion.*$$")) {
+            addShimUrl(jar)
+            logInfo(s"include streampark lib: $jarName")
+          }
         }
       }
     })
@@ -160,7 +162,8 @@ object FlinkShimsProxy extends Logger {
         new ChildFirstClassLoader(
           shimsUrls.toArray,
           Thread.currentThread().getContextClassLoader,
-          getFlinkShimsResourcePattern(flinkVersion.majorVersion))
+          getFlinkShimsResourcePattern(flinkVersion.majorVersion)
+        )
       })
   }
 

--- a/streampark-flink/streampark-flink-proxy/src/main/scala/org/apache/streampark/flink/proxy/FlinkShimsProxy.scala
+++ b/streampark-flink/streampark-flink-proxy/src/main/scala/org/apache/streampark/flink/proxy/FlinkShimsProxy.scala
@@ -86,7 +86,7 @@ object FlinkShimsProxy extends Logger {
 
         // 3) add only streampark shims jar
         addShimsUrls(flinkVersion, file => {
-          if (file != null && file.getName.startsWith("streampark-flink-shims")) {
+          if (file.getName.startsWith(FLINK_SHIMS_PREFIX)) {
             shimsUrls += file.toURI.toURL
           }
         })

